sorting/strand_sort.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file strand_sort.cpp
+ * @brief Implementation of [Strand Sort](https://en.wikipedia.org/wiki/Strand_sort) algorithm.
+ *
+ * @details
+ * Strand Sort is a sorting algorithm that works in (O(n)) time if list is already sorted and works in (O(n * n)) in worst case.
+ * 
+ * ### Algorithm
+ * + It is passed over the array to be sorted once and the ascending (sequential) numbers are taken.
+ * + After the first iteration, the sequential sub-array is put on the empty sorted array.
+ * + The main sequence is passed over again and a new sub-sequence is created in order.
+ * + Now that the sorted array is not empty, the newly extracted substring is merged with the sorted array.
+ * + Repeat types 3 and 4 until the sub-sequence and main sequence are empty.
+ * 
+ * @author [Mertcan Davulcu](https://github.com/mertcandav)
+ */
+#include <iostream>
+#include <iostream>
+#include <list>
+
+namespace strand {
+    // Element type of list.
+    template <typename T>
+    /*
+    * Apply sorting.
+    * lst: List to sorted.
+    */
```

Comment:
You can also provide further information by using
`@details` as in the first comment block. 🙂

See #962 as a reference for how to document functions.
```suggestion
    /**
    * @brief Apply sorting
    * @param lst List to be sorted
    * @returns <provide description here>
    */
```